eva-tools/src/main/java/embl/ebi/variation/eva/vcfdump/BiodataVariantToVariantContextConverter.java

```diff
@@ -0,0 +1,183 @@
+import org.opencb.biodata.models.variant.VariantSourceEntry;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Created by pagarcia on 27/06/2016.
+ */
+public class BiodataVariantToVariantContextConverter {
+
+    public static final String GENOTYPE_KEY = "GT";
+    private final VariantContextBuilder variantContextBuilder;
+    private final List<VariantSource> sources;
+    private CellbaseWSClient cellbaseClient;
+    private String regionSequence;
+    private Map<String, Map<String,String>> studiesSampleNamesEquivalences;
+    private static final int NO_CALL_ALLELE_INDEX = 2;
+
+    public BiodataVariantToVariantContextConverter(List<VariantSource> sources, CellbaseWSClient cellbaseWSClient,
+                                                   Map<String, Map<String,String>> studiesSampleNamesEquivalences)
+    {
+        this.sources = sources;
+        this.cellbaseClient = cellbaseWSClient;
+        this.studiesSampleNamesEquivalences = studiesSampleNamesEquivalences;
+        variantContextBuilder = new VariantContextBuilder();
+    }
+
+    public VariantContext transform(Variant variant, Region region) throws CellbaseSequenceDownloadError {
+        String[] allelesArray = getAllelesArray(variant, region);
+
+        Set<Genotype> genotypes = getGenotypes(variant, allelesArray);
+
+        VariantContext variantContext = variantContextBuilder
+                .chr(variant.getChromosome())
+                // TODO: check start and end for indels
+                .start(variant.getStart())
+                .stop(variant.getEnd())
+                .noID()
+                .alleles(allelesArray)
+                .unfiltered()
+                .genotypes(genotypes).make();
+        return variantContext;
+    }
+
+    private String[] getAllelesArray(Variant variant, Region region) throws CellbaseSequenceDownloadError {
+        String[] allelesArray;
+        // if there are indels, we cannot use the normalized alleles, (hts forbids empty alleles) so we have to take them from cellbase
+        if (variant.getReference().isEmpty() || variant.getAlternate().isEmpty()) {
+            String contextNucleotide;
+            if (region != null) {
+                contextNucleotide = getContextNucleotideFromCellbaseCachingRegions(variant, variant.getStart(), region);
+            } else {
+                contextNucleotide = getContextNucleotideFromCellbase(variant, variant.getStart());
+            }
+
+            allelesArray = new String[] {contextNucleotide + variant.getReference(), contextNucleotide + variant.getAlternate()};
+            variant.setEnd(variant.getStart() + allelesArray[0].length() - 1);
+        } else {
+            allelesArray = new String[] {variant.getReference(), variant.getAlternate()};
+        }
+
+        return allelesArray;
+    }
+
+    private String getContextNucleotideFromCellbase(Variant variant, Integer start) throws CellbaseSequenceDownloadError {
```

Comment:
We could check if there is any Ensembl WS por getting sequences also. Anyway, I like adding the submitted alleles to the database, because the dumper will be more efficient (won't do any calls to external WS). In the other hand, the database would be bigger (but not so much, adding two chars per variant seems insignificant) and we woundn't be able to use opencga (we should fork it and make the desired changes or move to a different storage).